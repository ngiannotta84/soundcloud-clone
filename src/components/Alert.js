import React from "react";
import PropTypes from "prop-types";
import "../styles/alert.css";

const Alert = ({ message, success }) => {
  if (!message) return null;

  return (
    <p className={`alert alert-${success ? "success" : "error"}`}>{message}</p>
  );
};

Alert.defaultProps = {
  success: false,
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

export default Alert;
