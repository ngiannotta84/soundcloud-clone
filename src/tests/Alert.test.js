import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

xdescribe("Alert", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Alert message="" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays an error message", () => {
    const { asFragment } = render(<Alert message="Error!" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a success message", () => {
    const { asFragment } = render(<Alert message="Success!!!!" success />);

    expect(asFragment()).toMatchSnapshot();
  });
});
