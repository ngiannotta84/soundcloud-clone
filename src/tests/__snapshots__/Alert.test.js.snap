// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays a success message 1`] = `
<DocumentFragment>
  <p
    class="alert alert-success"
  >
    Success!!!!
  </p>
</DocumentFragment>
`;

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <p
    class="alert alert-error"
  >
    Error!
  </p>
</DocumentFragment>
`;

exports[`Alert renders correctly 1`] = `<DocumentFragment />`;
